path_roles.go

```diff
@@ -166,7 +166,7 @@ func (b *azureSecretBackend) pathRoleUpdate(ctx context.Context, req *logical.Re
 		role.AzureRoles = parsedRoles
 	}
 
-	roleIDs := make(map[string]bool)
+	roleSet := make(map[azureRole]bool)
```

Comment:
```
type azureRole struct {
	RoleName string `json:"role_name"` // e.g. Owner
	RoleID   string `json:"role_id"`   // e.g. /subscriptions/e0a207b2-.../providers/Microsoft.Authorization/roleDefinitions/de139f84-...
	Scope    string `json:"scope"`     // e.g. /subscriptions/e0a207b2-...
}
```
I notice that the PR says it wants RoleID and Scope to be unique, but actually, if I'm reading this right, _any of_ these three fields could be different and it would result in an azureRole not being equal to any already in the map. Is that what you want? Also, any fields added in the future could change behavior around equality comparisons. Would it make more sense to create a custom equality comparison to lock down the behaviour?

Reply:
I probably could have been clearer in the PR. I did mean the combo of RoleID+Scope needs to be unique, so the current behavior works (RoleName is effectively joined to RoleID and isn't a unique variable). But your comment about expansion makes sense. I think I'll use a key derived from the struct so changes don't message things up.